<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="GazeboSystem" type="system">
        <xacro:if value="$(eval is_sim)">
            <hardware>
               <plugin>gz_ros2_control/GazeboSimSystem</plugin>
            </hardware>
        </xacro:if>

        <xacro:if value="$(eval not is_sim)">
            <hardware>
                <plugin>bumperbot_controller/RobotInterface</plugin>
                <param name="port">/dev/ttyACM0</param>
            </hardware>
        </xacro:if>
        <!-- ******************************************* lb -->

        <joint name="lb1_base">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="lb1_2">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="lb2_3">
            <command_interface name="position">
                <param name="min">-0.523</param>
                <param name="max">2.61</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <!-- ******************************************* lf -->
         <joint name="lf1_base">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="lf1_2">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="lf2_3">
            <command_interface name="position">
                <param name="min">-0.523</param>
                <param name="max">2.61</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <!-- ******************************************* rb -->
         <joint name="rb1_base">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="rb1_2">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="rb2_3">
            <command_interface name="position">
                <param name="min">-0.523</param>
                <param name="max">2.61</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- ******************************************* rf -->
         <joint name="rf1_base">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="rf1_2">
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="rf2_3">
            <command_interface name="position">
                <param name="min">-0.523</param>
                <param name="max">2.61</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>

   <gazebo>
       <plugin filename="gz_ros2_control-system" name="gz_ros2_control::GazeboSimROS2ControlPlugin">
            <!-- <update_rate>100.0</update_rate> -->
            <parameters>$(find quadruped_description)/config//my_controllers.yaml</parameters> 
        </plugin>
    </gazebo>

</robot>